modules/route_monitor.cpp

```diff
@@ -0,0 +1,460 @@
+/*
+ * Copyright (c) 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <znc/IRCNetwork.h>
+#include <znc/IRCSock.h>
+#include <znc/Modules.h>
+
+namespace {
+
+// SplitAndSend constructs commands and replies <= 512 octets, when the last
+// parameter is a comma-separated list.
+//
+// Precondition: The length of base plus the length of any individual member of
+// the list must be <= 512.
+template <typename Iterator, typename ConvertToCStringFn>
```

Comment:
Is this template needed?

Reply:
Iterator is needed since SplitAndSend is called with a variety of container iterators (std::set<CasefoldedNick>, std::vector<CString>,  std::set<CString>).

ConvertToCStringFn also needs to be templated since it can be called with a CasefoldedNick or a CString.